### PR TITLE
[BottomAppBar] clean up how bottom app bar path renders to avoid using hardcoded values

### PR DESCRIPTION
**Pull request by fluxminds**
Both floating button and the path cut depended on many position constants to decide their position. This change removes the dependency of the path cut on those constant since the path cut should follow the floating button's realtime position.

Before the change:
simulator screen shot - iphone x - 2018-09-18 at 10 47 56

After the change:
simulator screen shot - iphone x - 2018-09-18 at 10 46 46

### CODE REVIEW

File: components/BottomAppBar/src/MDCBottomAppBarView.m [hunk 1]
```diff
@@ -56,16 +56,14 @@ @interface MDCBottomAppBarView () <CAAnimationDelegate>
 @implementation MDCBottomAppBarView
 
 - (instancetype)initWithFrame:(CGRect)frame {
-  self = [super initWithFrame:frame];
-  if (self) {
+  if (self = [super initWithFrame:frame]) {
```
Comment (jverkoey) [inline components/BottomAppBar/src/MDCBottomAppBarView.m]: The previous style is [our repo's defacto style](https://github.com/material-components/material-components-ios/search?q=%22if+%28self%29+%7B%22&unscoped_q=%22if+%28self%29+%7B%22), please revert these changes.

File: components/BottomAppBar/src/MDCBottomAppBarView.m [hunk 2]
```diff
@@ -361,8 +358,8 @@ - (void)setFloatingButtonHidden:(BOOL)floatingButtonHidden animated:(BOOL)animat
     }];
   } else {
     _floatingButton.hidden = NO;
-    [self cutBottomAppBarViewAnimated:animated];
     [_floatingButton expand:animated completion:nil];
+    [self cutBottomAppBarViewAnimated:animated];
```
Comment (jverkoey) [inline components/BottomAppBar/src/MDCBottomAppBarView.m]: Why did we need to change the order here?

**Review by jverkoey — CHANGES_REQUESTED** [review]
Please provide before/after screenshots in order to verify that nothing changed.

File: components/BottomAppBar/src/private/MDCBottomAppBarLayer.m [hunk 3]
```diff
@@ -36,228 +36,81 @@ + (instancetype)layer {
...
-                        width:width
-                       height:height
-                   startAngle:kStartAngle
-                     endAngle:kEndAngle];
-      } else {
-        [self pathWithCutRight:bottomBarPath
-                         width:width
-                        height:height
-                    startAngle:kStartAngle
-                      endAngle:kEndAngle];
```
Comment (jverkoey) [inline components/BottomAppBar/src/private/MDCBottomAppBarLayer.m:108]: Awesome - it looks like you made the path logic more generic. Please include a summary of this work in the PR's description.

File: components/BottomAppBar/src/MDCBottomAppBarView.m (hunk 1 above)
Comment (fluxminds) [reply components/BottomAppBar/src/MDCBottomAppBarView.m]: Sure. Addressed in f4423aa30b94ed3b92a7868fcebfcfdcac58b3d7.

File: components/BottomAppBar/src/MDCBottomAppBarView.m (hunk 2 above)
Comment (fluxminds) [reply components/BottomAppBar/src/MDCBottomAppBarView.m]: It doesn't affect the behavior. Simple because I saw `healBottomAppBarViewAnimated` was called before `[_floatingButton collapse]` on the healing part. So in the cutting part, I thought it should expand first as the reverse way. The behavior doesn't change.

Comment (jverkoey) [reply components/BottomAppBar/src/MDCBottomAppBarView.m]: If we can revert this change without affecting the PR let's do so in the spirit of minimizing deltas.

**Review by jverkoey — APPROVED** [review]
LGTM with one small request to revert a change that may not be necessary for this PR.

File: components/BottomAppBar/src/private/MDCBottomAppBarLayer.m (hunk 3 above)
Comment (fluxminds) [reply components/BottomAppBar/src/private/MDCBottomAppBarLayer.m:108]: Summary added in the description, and added the screenshots. Sorry for missing those because I didn't think about the split PR should follow the rules strictly. Thanks for reminding it!

File: components/BottomAppBar/src/MDCBottomAppBarView.m (hunk 2 above)
Comment (fluxminds) [reply components/BottomAppBar/src/MDCBottomAppBarView.m]: Sure. Reverted in 4961552806e27310659c898640fd7e64822f6e29. Will merge after CI is finished.
